backend/modules/database/db_queries.bal

```diff
@@ -86,12 +86,109 @@ isolated function upsertFavouritesQuery(string email, int appId, boolean isFavou
     ON DUPLICATE KEY UPDATE
         is_favourite = ${isFavourite}`;
 
-# Build query to check if an application ID is valid and active.
+# Build query to fetch app details with filters for validation and admin operations.
 #
-# + appId - Application ID to validate
-# + return - Parameterized SQL query that returns boolean result
-isolated function isValidAppIdQuery(int appId) returns sql:ParameterizedQuery => `
-    SELECT EXISTS(
-        SELECT 1 FROM apps 
-        WHERE id = ${appId} AND is_active = 1
-    ) AS is_valid`;
+# + filters - Filter criteria to apply when querying apps
+# + return - Parameterized SQL query with applied filters
+isolated function fetchAppByFilterQuery(AppFilters filters) returns sql:ParameterizedQuery {
+    sql:ParameterizedQuery mainQuery = `
+        SELECT 
+            a.id,
+            a.header,
+            a.url,
+            a.description,
+            a.version_name AS versionName,
+            a.tag_id AS tagId,
+            a.icon,
+            a.added_by AS addedBy,
+            a.updated_by,
+            t.name AS tagName,
+            t.color as tagColor,
+            t.is_active
+        FROM apps a
+        LEFT JOIN tags t ON a.tag_id = t.id
+    `;
+
+    sql:ParameterizedQuery[] filterQueries = [];
+    if filters.header is string {
+        filterQueries.push(` a.header = ${filters.header}`);
+    }
+
+    if filters.id is int {
+        filterQueries.push(` a.id = ${filters.id}`);
+    }
+
+    if filters.url is string {
+        filterQueries.push(` a.url = ${filters.url}`);
+    }
+
+    if filters.addedBy is string {
+        filterQueries.push(` a.added_by = ${filters.addedBy}`);
+    }
+
+    if filters.isActive is string {
+        filterQueries.push(` a.is_active = ${filters.isActive}`);
+    }
+
+    mainQuery = buildSqlSelectQuery(mainQuery, filterQueries);
+
+    return mainQuery;
+}
+
+# Build query to create a new app.
+# + app - App data to insert
+# + return - Parameterized query for app creation
+isolated function createAppQuery(CreateApp app) returns sql:ParameterizedQuery {
+    string userGroupsCsv = app.userGroups.length() > 0 ? string:'join(",", ...app.userGroups) : "";
```

Comment:
[ballerinax/mysql](https://central.ballerina.io/ballerinax/mysql/latest) library contains `sql:queryConcat()` and `sql:arrayFlattenQuery()` util functions which make it easier to create a dynamic/constant complex queries. You can use `sql:arrayFlattenQuery()` here as userGroups is a string[].

e.g. 
```
int[] ids = [1, 2];
sql:ParameterizedQuery sqlQuery = sql:queryConcat(`
    SELECT Name, Department, Country 
    FROM DataTable 
    WHERE id IN (`, sql:arrayFlattenQuery(ids), `)
`);
```